ceno_emul/src/addr.rs

```diff
@@ -197,17 +197,17 @@ impl ops::AddAssign<u32> for ByteAddr {
 }
 
 pub trait IterAddresses {
-    fn iter_addresses(&self) -> impl Iterator<Item = Addr>;
+    fn iter_addresses(&self) -> impl ExactSizeIterator<Item = Addr>;
```

Comment:
Interesting!  Why do we need the additional guarantees?

(I wonder if we should also add `+ DoubleEndedIterator`, or perhaps just give a concrete type?  But that's just idle musing.)

Reply:
For `len()`